### ISSUE TITLE
[Bug]: Replay buffer: change psutil.virtualmemory() from available to total

### ISSUE THREAD

**Issue by antoniopioricciardi** [DLR-RM/stable-baselines3#1839]
### 🐛 Bug

In `stable_baselines3.common/buffers.py`, line 215, we check whether `total_memory_usage > mem_available`, but `mem_available` is assigned as:
`mem_available = psutil.virtual_memory().available`

This command, however, only returns the memory **currently available** . Therefore, in modern systems that tend to take all the ram they can, that command will return a very low value (I even tested for systems with 64 gb of ram, still gives same issue with different value for mem_available).

Is there a reason for using `psutil.virtual_memory().available` instead of ` psutil.virtual_memory().total` ?

I mean, I know that people could have other memory intensive processes running on their computer, but I think leaving the freedom of choice to users would be the best option. Thank you

### To Reproduce

```python
import torch
from stable_baselines3.common.buffers import ReplayBuffer
from gym.spaces import Box
from gym.spaces import Discrete

device = ('cuda' if torch.cuda.is_available() else 'cpu')
rb = ReplayBuffer(
    10000,
    Box(0.0, 1.0, (4, 84, 84), dtype=float),
    Discrete(4),
    device,
    handle_timeout_termination=True,
)
```

### Relevant log output / Error message

```shell
/Users/user/opt/anaconda3/envs/relative_rl/lib/python3.9/site-packages/stable_baselines3/common/buffers.py:219: 
UserWarning: This system does not have apparently enough memory to store the complete replay buffer 4.52GB > 0.95GB
```

### System Info

MacOS: 14.2.1 (23C71)
System RAM: 16 GB
M1 Pro chip
python 3.9.17
gym: 0.23.1
gymnasium: 0.29.1

----

Ubuntu 20.04.4 LTS
System RAM: 64 GB
Nvidia RTX 3090
python 3.9.18
gym: 0.23.1
gymnasium: 0.29.1

Library installed with pip

### Checklist

- [X] My issue does not relate to a custom gym environment. (Use the custom gym env template instead)
- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> This command, however, only returns the memory currently available . Therefore, in modern systems that tend to take all the ram they can, that command will return a very low value (I even tested for systems with 64 gb of ram, still gives same issue with different value for mem_available).

If the RAM is not available, why would you use `psutil.virtual_memory().total`?
If you fill up the replay buffer, it will end up in the swap and then freeze the computer if there is not enough memory available, no?

Anyway, it is only a warning and doesn't prevent the training to continue, right?

**Comment by antoniopioricciardi** [comment]
Because usually the system takes care of freeing up the space. For example, the Linux computer is sitting idle but `psutil.virtual_memory().available` still returns only 3.54GB.

However, you're totally right. It is just a warning and does not prevent the code from running. My issue was with a line of my code that, because of that warning, wasn't showing up somehow.

In fact, after changing it to `psutil.virtual_memory().total` I was able to see where the error was and now, even reverting it to `psutil.virtual_memory().available` the code works fine. 

Sorry for the useless issue and thank you!
